File: source codes/Lab 12/Assigngrades.cs
using System;
using System.Collections.Generic;
using System.Linq;

class MainClass {
  public static void Main() {
    int size,count=0,counter=0,test;
    string lettergrade;
    Console.WriteLine("What is the class size: ");
    size = Convert.ToInt32(Console.ReadLine());
    int[] Class = new int[size];
    Console.WriteLine("Enter their grades: ");
    while (count < size){
      test = Convert.ToInt32(Console.ReadLine());
      if (test >= 0 && test <= 100) {
        Class[count] = test;
        count++;
      }
      else
        {Console.WriteLine("Invalid number, try again.");}
    }
  while (counter < size){
    if(Class[counter] <=100 && Class[counter] >=90) 
      {lettergrade = "A";}
    else if(Class[counter] <= 89 && Class[counter] >=80) 
      {lettergrade = "B";}
    else if(Class[counter] <= 79 && Class[counter] >=70) 
      {lettergrade = "C";}
    else if(Class[counter] <= 69 && Class[counter] >=60) 
      {lettergrade = "D";}
    else 
      {lettergrade = "F";}
    Console.WriteLine("Student " + counter + " score is " + Class[counter] + " and grade is " + lettergrade);
    counter++;
    }
  }
}

File: School source codes/Lab 12/Assigngrades.rb
count = 0
counter = 0
print "What is the Class size: "
size = gets.chomp.to_i
Classe = Array.new(size)
print "Enter their grades: "
while (count < size)
  test = gets.chomp.to_i
  if (test >= 0 && test <= 100)
    Classe[count] = test
    count += 1
  else
    puts "Invalid number, try again."
  end
end
while (counter < size)
  if(Classe[counter] <=100 && Classe[counter] >=90) 
    lettergrade = "A"
  elsif(Classe[counter] <= 89 && Classe[counter] >=80) 
    lettergrade = "B"
  elsif(Classe[counter] <= 79 && Classe[counter] >=70) 
    lettergrade = "C"
  elsif(Classe[counter] <= 69 && Classe[counter] >=60) 
    lettergrade = "D"
  else 
    lettergrade = "F"
  end
  puts "Student number #{counter+1}'s score is #{Classe[counter]} and their grade is an #{lettergrade}"
  counter += 1
end